src/wish.cpp

```diff
@@ -594,11 +594,11 @@ void debug_menu::wishitem( player *p, const tripoint &pos )
                     if( granted.count_by_charges() ) {
                         if( amount > 0 ) {
                             granted.charges = amount;
-                            p->i_add( granted );
+                            get_map().add_item_or_charges( p->pos(), granted );
```

Comment:
instead of always spawning on the floor, do something like this instead:
```suggestion
                            if( p->can_stash( granted ) {
                              p->i_add( granted );
                            } else {
                              get_map().add_item_or_charges( p->pos(), granted );
                            }
```
same for the other spot.